src/lib.rs

```diff
@@ -137,6 +138,48 @@ impl AtomicF64 {
     }
 }
 
+/// Utility functions for arguments sanity check
+pub(crate) fn assert_is_valid_sample_rate(sample_rate: SampleRate) {
+    let sample_rate = sample_rate.0;
+    // allow arbitrary sample rates in tests
+    if cfg!(test) {
+        if sample_rate == 0 {
+            panic!(
+                "NotSupportedError - Invalid sample rate: {:?} is negative or zero (test mode)",
+                sample_rate
+            );
+        }
+    } else {
+        // An implementation MUST support sample rates in at least the range 8000 to 96000
+        // @note - `test/media.rs` relies on `RENDER_QUANTUM_SIZE` this should be
+        // cleaned together with the MediaElement, see pull #106
+        if sample_rate < u32::try_from(RENDER_QUANTUM_SIZE).unwrap() || sample_rate > 96000 {
+            panic!(
+                "NotSupportedError - Invalid sample rate: {:?} is outside range [8000, 96000]",
+                sample_rate
+            );
+        }
+    }
+}
+
+pub(crate) fn assert_is_valid_number_of_channels(number_of_channels: usize) {
+    if number_of_channels == 0 || number_of_channels > MAX_CHANNELS {
+        panic!(
+            "NotSupportedError - Invalid number of channels: {:?} is outside range [1, 32]",
```

Comment:
Can you replace the harcoded `32` with `{}` and a `MAX_CHANNELS` argument?
In theory, a later version could allow for more channels of course (either by forking and recompiling, or by introducing const generics ourselves)

Reply:
right, it's fixed